test/jasmine/tests/plot_promise_test.js

```diff
@@ -488,7 +492,34 @@ describe('Plotly.___ methods', function() {
             }, function(err) {
                 expect(err).toBeDefined();
                 expect(err.message).toBe('Resize must be passed a displayed plot div element.');
-            }).then(done);
+            })
+            .catch(failTest)
+            .then(done);
+        });
+
+        xit('should return a resolved promise and purge plotly attributes - tmp', function(done) {
```

Comment:
@alexcjohnson you would have to enable it to see the error

Reply:
Playing with this a little... I think this version of the test *is* correct, and what's needed is just to interpret the comment in the code `// return if there is nothing to resize` a little more broadly - if the plot has been purged (in which case it has no `layout`) that's another "nothing to resize." So perhaps:
```js
if(!gd.layout || (gd.layout.width && gd.layout.height)) {
```
Which, at least if I run it in the console (`Plotly.Plots.resize(gd);Plotly.purge(gd);`) successfully avoids throwing an error.

Hmm, it also occurs to me, looking at the code, we should bail out if `isHidden(gd)` as well, ie when `resize` was called the plot was visible, but before `_redrawTimer` fired it has been hidden/removed. That's yet another "nothing to resize." So I guess:
```js
if(!gd.layout || (gd.layout.width && gd.layout.height) || isHidden(gd)) {
```
The only other test we do outside `_redrawTimer` is `!gd`, but I don't see any way `gd` could be truthy before and falsy 100ms later, it's still going to be a DOM element, the worst that can happen is it gets detached; so I think that should do it!